docker/Dockerfile.build

```diff
@@ -0,0 +1,61 @@
+# syntax = docker/dockerfile:1.4
+FROM bitnami/minideb:latest as main
```

Comment:
Are we certain that `latest` will always be safe? Also are we okay with taking a dependency on a corporation?

Reply:
This image was super tiny, which is why I used it. I can play around with a few others that don't have the corporate dependency, and fix the version so its predictable.

Reply:
Using `rust:1-slim-bullseye` as the root image I was able to get a very similar build image size.